src/Paket.Core/Utils.fs

```diff
@@ -211,27 +220,67 @@ let ProgramFilesX86 =
 /// The system root environment variable. Typically "C:\Windows"
 let SystemRoot = Environment.GetEnvironmentVariable "SystemRoot"
 
+let isMonoRuntime =
+    not (Object.ReferenceEquals(Type.GetType "Mono.Runtime", null))
+
 /// Determines if the current system is an Unix system
-let isUnix = int Environment.OSVersion.Platform |> fun p -> (p = 4) || (p = 6) || (p = 128)
+let isUnix = 
+#if NETSTANDARD1_6
+    System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(
+        System.Runtime.InteropServices.OSPlatform.Linux) || 
+    System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(
+        System.Runtime.InteropServices.OSPlatform.OSX)
+#else
+    int Environment.OSVersion.Platform |> fun p -> (p = 4) || (p = 6) || (p = 128)
+#endif
 
 /// Determines if the current system is a MacOs system
 let isMacOS =
+#if NETSTANDARD1_6
+    System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(
+        System.Runtime.InteropServices.OSPlatform.OSX)
+#else
     (Environment.OSVersion.Platform = PlatformID.MacOSX) ||
         // osascript is the AppleScript interpreter on OS X
         File.Exists "/usr/bin/osascript"
+#endif
 
 /// Determines if the current system is a Linux system
-let isLinux = isUnix && not isMacOS
+let isLinux = 
+#if NETSTANDARD1_6
+    System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(
+        System.Runtime.InteropServices.OSPlatform.Linux)
+#else
+    isUnix && not isMacOS
+#endif
 
 /// Determines if the current system is a Windows system
 let isWindows =
+#if NETSTANDARD1_6
+    System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(
+        System.Runtime.InteropServices.OSPlatform.Windows)
+#else
     match Environment.OSVersion.Platform with
     | PlatformID.Win32NT | PlatformID.Win32S | PlatformID.Win32Windows | PlatformID.WinCE -> true
     | _ -> false
+#endif
+
 
 /// Determines if the current system is a mono system
 /// Todo: Detect mono on windows
-let isMono = isUnix
+let isMono = 
+#if NETSTANDARD1_6
+    false
```

Comment:
Sounds good

Am 03.09.2016 12:58 nachm. schrieb "Matthias Dittrich" <
notifications@github.com>:

> In src/Paket.Core/Utils.fs
> <https://github.com/fsprojects/Paket/pull/1785#discussion_r77435342>:
>
> >
> >  /// Determines if the current system is a mono system
> >  /// Todo: Detect mono on windows
> > -let isMono = isUnix
> > +let isMono =
> > +#if NETSTANDARD1_6
> > +    false
>
> Maybe a good way is to mark them all as obsolete and replace them one by
> one in the future?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/fsprojects/Paket/pull/1785/files/38cc48c4cb6da1467608e98487c9cc9000ec2c9d#r77435342>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AADgNASVB_VIsnAQ4AX8EguwNO8SNJQWks5qmVLvgaJpZM4JDrgC>
> .
>